src/CommandFormatter.php

```diff
@@ -0,0 +1,130 @@
+
+            return $priorityA === $priorityB ? strcmp($a, $b) : $priorityA - $priorityB;
+        });
+
+        return $command;
+    }
+
+    /**
+     * Flatten API command's nested arrays for easier handling
+     * @param array<mixed> $cmd API Command
+     * @return array<mixed>
+     */
+    public static function flattenCommand($cmd)
+    {
+        $newcmd = [];
+        foreach ($cmd as $key => $val) {
+            if (isset($val)) {
+                $val = preg_replace("/\r|\n/", "", $val);
+                $newKey = \strtoupper($key);
+                if (is_array($val)) {
+                    foreach ($cmd[$key] as $idx => $v) {
+                        $newcmd[$newKey . $idx] = $v;
+                    }
+                } else {
+                    $newcmd[$newKey] = $val;
+                }
+            }
+        }
+
+        // Sort the command array based on priority
+        return self::getSortedCommand($newcmd);
+    }
+
+    /**
+     * Format the command array into a plain text string
+     *
+     * @param array<string, string> $command The command array to be formatted
+     * @return string The formatted command string
+     */
+    public static function formatCommand(array $command): string
+    {
+        $tmp = "";
+
+        foreach ($command as $key => $val) {
+            $tmp .= "$key = $val\n";
+        }
+        return $tmp;
+    }
+
+    /**
+     * Get the priority array with regex patterns
+     *
+     * @return array<string, int> The priority array
+     */
+    private static function getPriorityArray(): array
+    {
+        $priority = [
+            "COMMAND" => 1,
+            "/^(DOMAIN|DNSZONE|NAMESERVER|ZONE)[0-9]*$/" => 2,
+            "/^(PERIOD|ACTION|AUTH|TARGET|X-FEE-COMMAND|RENEWALMODE|LIMIT|WIDE)$/" => 3,
+            "/^(TRANSFERLOCK|DNSSEC0|X-FEE-AMOUNT|LOG|TYPE|OBJECT|INACTIVE|OBJECTID|OBJECTCLASS|ORDER|ORDERBY|CURRENCYFROM|CURRENCYTO)$/" => 4,
+        ];
+
+        $contactTypes = [
+            "OWNERCONTACT" => 5,
+            "ADMINCONTACT" => 6,
+            "TECHCONTACT" => 7,
+            "BILLINGCONTACT" => 8
+        ];
+        $contactFields = [
+            "FIRSTNAME" => 1,
+            "LASTNAME" => 2,
+            "ORGANIZATION" => 3,
+            "STREET" => 4,
+            "ZIP" => 5,
+            "CITY" => 6,
+            "STATE" => 7,
+            "COUNTRY" => 8,
+            "PHONE" => 9,
+            "EMAIL" => 10,
+            "CONTACT" => 11
+        ];
+
+        foreach ($contactTypes as $type => $typePriority) {
+            foreach ($contactFields as $field => $fieldPriority) {
+                $priority["/^{$type}[0-9]+{$field}$/"] = ($typePriority * 10) + $fieldPriority;
+            }
+        }
+
+        return $priority;
+    }
+
+    /**
+     * Find the priority of a given key
+     *
+     * @param string $key The key to find the priority for
+     * @param array<string, int> $priority The priority array
```

Comment:
Please avoid blank space in the parameter's type declaration. I noticed that this doesn't work well with PHPStan.

`@param array<string, int> $priority The priority array`
vs.
`@param array<string,int> $priority The priority array`

Idea: Feel free to grep for all `@param array<string, ` matches and replace them with `@param array<string,`
I could imagine that you'll patch further cases by that. See the other method before as example.

ty